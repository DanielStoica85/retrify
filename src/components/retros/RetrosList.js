import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import Retro from './Retro';

const RetrosList = props => {
    const { retros } = props;
    const retrosList =
        retros &&
        retros.map(retro => {
            return <Retro retro={retro} key={retro.id} />;
        });

    return (
        <div className="retros-list section">
            <p>Retros</p>
            {retros.length > 0 ? (
                <p>{retrosList}</p>
            ) : (
                <p>
                    You currently have no retros. Click{' '}
                    <Link to="/retro">HERE</Link> to add your first retro.
                </p>
            )}
        </div>
    );
};

RetrosList.propTypes = {
    retros: PropTypes.array
};

export default RetrosList;
