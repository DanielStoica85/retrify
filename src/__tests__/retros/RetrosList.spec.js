import React from 'react';
import renderer from 'react-test-renderer';
import RetrosList from '../../components/retros/RetrosList';

describe('RetrosList component', () => {
    const props = {
        retros: [
            {
                title: 'Some title',
                author: 'Some author',
                date: 'May 10th, 2019',
                id: 1
            },
            {
                title: 'Some title',
                author: 'Some author',
                date: 'May 10th, 2019',
                id: 2
            },
            {
                title: 'Some title',
                author: 'Some author',
                date: 'May 10th, 2019',
                id: 3
            }
        ]
    };
    it('renders correctly', () => {
        const component = renderer.create(<RetrosList {...props} />).toJSON();
        expect(component).toMatchSnapshot();
    });
});
