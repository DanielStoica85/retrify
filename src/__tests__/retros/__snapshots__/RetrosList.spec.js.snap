// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RetrosList component renders correctly 1`] = `
<div
  className="retros-list section"
>
  <div>
    <div
      className="mt-3 card"
    >
      <div
        className="card-body"
      >
        <div
          className="card-title"
        >
          Some title
        </div>
        <div
          className="card-subtitle"
        >
          Added by 
          Some author
        </div>
      </div>
      <div
        className="card-footer"
        style={
          Object {
            "backgroundColor": "lightgray",
            "borderColor": "lightgray",
          }
        }
      >
        May 10th, 2019
      </div>
    </div>
  </div>
  <div>
    <div
      className="mt-3 card"
    >
      <div
        className="card-body"
      >
        <div
          className="card-title"
        >
          Some title
        </div>
        <div
          className="card-subtitle"
        >
          Added by 
          Some author
        </div>
      </div>
      <div
        className="card-footer"
        style={
          Object {
            "backgroundColor": "lightgray",
            "borderColor": "lightgray",
          }
        }
      >
        May 10th, 2019
      </div>
    </div>
  </div>
  <div>
    <div
      className="mt-3 card"
    >
      <div
        className="card-body"
      >
        <div
          className="card-title"
        >
          Some title
        </div>
        <div
          className="card-subtitle"
        >
          Added by 
          Some author
        </div>
      </div>
      <div
        className="card-footer"
        style={
          Object {
            "backgroundColor": "lightgray",
            "borderColor": "lightgray",
          }
        }
      >
        May 10th, 2019
      </div>
    </div>
  </div>
</div>
`;
